src/graph/optimizer/OptRule.h

```diff
@@ -30,6 +30,8 @@ class OptGroup;
 struct MatchedResult {
   const OptGroupNode *node{nullptr};
   std::vector<MatchedResult> dependencies;
+  // Boundary of the whole matched result, aka the dependencies of leaf node in MatchedResult.
+  std::vector<OptGroup *> boundary_;
```

Comment:
It looks like there is no need to save this variable.

Reply:
Yes, we could collect it in `match` method directly.